yb_migrate/cmd/importDataFileCommand.go

```diff
@@ -0,0 +1,264 @@
+}
+
+func parseFileTableMapping() {
+	if fileTableMapping != "" {
+		keyValuePairs := strings.Split(fileTableMapping, ",")
+		for _, keyValuePair := range keyValuePairs {
+			fileName, table := strings.Split(keyValuePair, ":")[0], strings.Split(keyValuePair, ":")[1]
+			tableNameVsFilePath[table] = filepath.Join(dataDir, fileName)
+		}
+	} else {
+		utils.PrintAndLog("Note: --file-table-map flag is not provided, default will assume the file names in format as mentioned in the docs. Refer - link")
+		// get matching file in data-dir
+		files, err := filepath.Glob(filepath.Join(dataDir, "*_data.sql"))
+		if err != nil {
+			utils.ErrExit("finding data files to import: %v", err)
+		}
+
+		if len(files) == 0 {
+			utils.ErrExit("No data files found to import in %q", dataDir)
+		} else {
+			utils.PrintAndLog("Table data files identified to import from data-dir are: %s", strings.Join(files, "\n"))
+		}
+
+		reTableName := regexp.MustCompile(`(\S+)_data.sql`)
+		for _, file := range files {
+			fileName := filepath.Base(file)
+
+			matches := reTableName.FindAllStringSubmatch(fileName, -1)
+			if len(matches) == 0 {
+				utils.ErrExit("datafile names in %q are not in right format, refer docs", dataDir)
+			}
+
+			tableName := matches[0][1]
+			tableNameVsFilePath[tableName] = file
+		}
+	}
+}
+
+func checkFileType() {
+	supported := false
+	for _, supportedFileType := range supportedFileTypes {
+		if fileType == supportedFileType {
+			supported = true
+			break
+		}
+	}
+
+	if !supported {
+		utils.ErrExit("given file-type %q is not supported", fileType)
+	}
+}
+
+func checkDataDirFlag() {
+	if dataDir == "" {
+		fmt.Fprintln(os.Stderr, `Error: required flag "data-dir" not set`)
+		os.Exit(1)
+	}
+	if !utils.FileOrFolderExists(dataDir) {
+		fmt.Fprintf(os.Stderr, "Directory: %s doesn't exists!!\n", dataDir)
+		os.Exit(1)
+	} else if dataDir == "." {
+		fmt.Println("Note: Using current working directory as data directory")
+	} else {
+		dataDir = strings.TrimRight(dataDir, "/")
+	}
+}
+
+func init() {
+	importDataCmd.AddCommand(importDataFileCmd)
+	registerCommonImportFlags(importDataFileCmd)
+
+	importDataFileCmd.Flags().StringVar(&fileType, "file-type", "csv",
+		"type of data file: csv, sql")
+
+	importDataFileCmd.Flags().StringVar(&delimiter, "delimiter", "\t",
+		"character used as delimiter in rows of the table(s)")
+
+	importDataFileCmd.Flags().StringVar(&dataDir, "data-dir", "",
+		"path to the directory contains data files to import into table(s)")
+	err := importDataFileCmd.MarkFlagRequired("data-dir")
+	if err != nil {
+		utils.ErrExit("mark 'data-dir' flag required: %v", err)
+	}
+
+	importDataFileCmd.Flags().StringVar(&fileTableMapping, "file-table-map", "",
+		"mapping between file name in 'data-dir' to table name.\n"+
+			"Note: default will assume the file names in format as mentioned in the docs. Refer - link") // TODO: if not given default should be file names
+
+	importDataFileCmd.Flags().BoolVar(&hasHeader, "has-header", false,
```

Comment:
@amit-yb why? default is there, user can give when needed? I  feel it will increase the work of adding one more flag.

Reply:
Users do not read the entire help text or documentation before they start using the tool. Unless the documentation is always using the `--has-header` flag, the users won't know about it. In this case, imagine the amount of work for the user if he tries a file import that has a header line.
-- The import is failed.
-- User is supposed to look at the log file where possibly he will get a data type mismatch error.
-- If he figures out that it is happening because of a header row, then he will have to rerun the command with the `--start-clean` option.

Reply:
Yes the import data file command will fail but user will not need to see logs, he will get it on the screen due to the `utils.ErrExit()` call.
Let's say the user has a file with a header but didn't give the `--has-header` flag then there can be cases:
1. If all the columns are not **varchar**: he will get the invalid syntax error for the first line itself as it will not following the row syntax based on data types used and command will exit.
2. If all columns are of varchar, then the header might also get considered as a row and will be pushed to the table.

The current default behavior is according to what we have in the copy command.